include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -6853,15 +6894,16 @@ namespace VectorTools
         p_d_triangulation
         = dynamic_cast<const parallel::distributed::Triangulation<dim,spacedim> *>(&dof.get_tria()))
       {
-        double my_values[2] = { mean, area };
-        double global_values[2];
+        std::complex<double> mean_double = mean;
+        double my_values[3] = { mean_double.real(), mean_double.imag(), area };
+        double global_values[3];
```

Comment:
Please add a comment explaining why you do this. For example something like
```
  The type used to store the elements of the global vector may be a real or a complex number. Do the global reduction always with real and imaginary types so that we don't have to distinguish, and to this end just copy everything into a complex number and, later, back into the original data type.
```